contrib/twmb/franz-go/franz/franz_test.go

```diff
@@ -0,0 +1,57 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package franz
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/twmb/franz-go/pkg/kgo"
+)
+
+func TestWrapClient(t *testing.T) {
+	client, err := kgo.NewClient()
+	assert.NoError(t, err)
+	wrapped := WrapClient(client)
+	assert.NotNil(t, wrapped)
+	assert.Equal(t, client, wrapped.Client)
+}
+
+func TestNewClient(t *testing.T) {
+	client, err := NewClient([]kgo.Opt{})
+	assert.NoError(t, err)
+	assert.NotNil(t, client)
+}
+
+func TestHeaders(t *testing.T) {
+	headers := []kgo.RecordHeader{
+		{Key: "key1", Value: []byte("value1")},
+		{Key: "key2", Value: []byte("value2")},
+	}
+	carrier := NewRecordHeadersCarrier(headers)
+
+	// Test ForeachKey
```

Comment:
For testing individual scenarios, use `t.Run()` to separate them out. This will make it easier to pinpoint what is going wrong if one of them were to fail.